package com.pratik.job.event;

import com.pratik.job.services.EmailNotificationService;
import lombok.RequiredArgsConstructor;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class NotificationKafkaConsumer {

    private final EmailNotificationService emailService;

    @KafkaListener(
            topics = "application.status.changed",
            groupId = "notification-service",
            containerFactory = "kafkaListenerContainerFactory"
    )

    public void handleStatusChanged(ApplicationStatusChangedEvent event) throws Exception {
        System.out.println("received ----------" + event);
        emailService.sendStatusChangedEmail(event);
    }

}
